triedb/pathdb/buffer.go

```diff
@@ -95,7 +95,12 @@ func (b *buffer) revertTo(db ethdb.KeyValueReader, nodes map[common.Hash]map[str
 		return nil
 	}
 	b.nodes.revertTo(db, nodes)
-	b.states.revertTo(accounts, storages)
+	// TODO(galaio): In order to be compatible with the legacy version, a temporary empty check is added,
+	// which may affect the reading result of pbss as flatReader, see: flatReader.Account()
+	// it could be removed in the future
+	if len(b.states.accountData) != 0 || len(b.states.storageData) != 0 {
```

Comment:
need add one more scenario which will lead to panic

Reply:
👍